feat(meshtls): initial guide

slonka opened the pull request:
closes https://github.com/kumahq/kuma-website/issues/1897

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits) yes 

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? yes

lahabana commented on app/_src/guides/progressively-rolling-in-strict-mtls.md:
```diff
@@ -0,0 +1,104 @@
+title: Progressively rolling in strict mTLS
+---
+
+The new [MeshTLS](/docs/{{ page.version }}/policies/meshtls/) policy allows you to gradually migrate services to mutual TLS without dropping a packet.
```
```suggestion
The [MeshTLS](/docs/{{ page.version }}/policies/meshtls/) policy allows you to gradually migrate services to mutual TLS without dropping a packet.
```

Docs should be intemporal

slonka commented:
> It would be good to have a mermaid diagram to show what things look like at each step

added, do you think I should also somehow show when things are in `strict` vs `permissive` mode?

> It would be good at the end to have a section: "Things to remember when migrating to strict TLS" with things like: "before turning something to strict check that this stat is no longer incrementing"

added that but need to check that metric... @johncowen - did we write down the metrics we want to use in the GUI to show TLS-ed / non TLS-ed requests / connections?

also do you have any other final points that we should add?

slonka commented:
@lahabana - probably shouldn't ask the users to calculate things by hand... maybe we should keep `tls_inspector` stats for now with a `*` that it's not 100% sure and point people to GUI once the feature linked by JC makes sense, WDYT?

lahabana commented:
Providing at least clarity as to which metrics users should like like seems ok to me no? What's being calculated here?

The graphs look good but would it be helpful to color the arrows depending on TLS and non-TLS traffic?

lahabana commented on app/_src/guides/progressively-rolling-in-strict-mtls.md:
```diff
@@ -0,0 +1,185 @@
+  - targetRef:
+      kind: Mesh
+    default:
+      mode: Permissive
```
I think in other guides we take stuff that can be directly copy pasted. Also I think the namespace should be `{{ site.namespace }}` e.g: https://github.com/kumahq/kuma-website/blob/master/app/_src/guides/gateway-builtin.md#define-a-route-using-meshhttproute

slonka replied:
yes, definitely!

lahabana reviewed (approved):
1 nit but LGTM. I've ran the guide and it worked.

However, the reliance on these stats is hard to comprehend as I show in this zoom clip:
https://konghq.zoom.us/clips/share/enLGWQHCv3HzEKlFWkEGEeuXdOcyDYdGTHu518KqCwf3z1ZnpuntGwhv2UO2CzLJPwkAxynTL04Uc6Bx-Js7xtxp.Cu8iOtgCRRiLZmBS

Is there anything we can do better there? If not should we just say we're going to improve this and link to the issue?

slonka commented on app/_src/guides/progressively-rolling-in-strict-mtls.md:
```diff
@@ -0,0 +1,185 @@
+kind: MeshTLS
+metadata:
+  name: redis
+  namespace: kuma-system
```
```suggestion
  namespace: {{ site.namespace }}
```
